From 0cda083fedc01f9bc2e1d09767f48b40c1bee64b Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:32:00 -0400
Subject: [scsi] hpsa: kill annoying messages about SSD Smart Path retries

Message-id: <1410186734-19986-70-git-send-email-thenzl@redhat.com>
Patchwork-id: 92091
O-Subject: [RHEL6.6 e-stor PATCH 69/83] hpsa: kill annoying messages about SSD Smart Path retries
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

There's nothing the user can or should do about these messages,
the commands are retried down the normal RAID path, and the
messages just flood the logs and sap performance.

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Reviewed-by: Joe Handzik <joseph.t.handzik@hp.com>
Reviewed-by: Mike Miller <michael.miller@canonical.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit c45166bee5831dc5ccc7d40e26ed758a92f0f940)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index a81e891..57e699b 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -1659,16 +1659,6 @@ static void process_ioaccel2_completion(struct ctlr_info *h,
  if (is_logical_dev_addr_mode(dev->scsi3addr) &&
   c2->error_data.serv_response ==
    IOACCEL2_SERV_RESPONSE_FAILURE) {
-  if (c2->error_data.status ==
-   IOACCEL2_STATUS_SR_IOACCEL_DISABLED)
-   dev_warn(&h->pdev->dev,
-    "%s: Path is unavailable, retrying on standard path.\n",
-    "HP SSD Smart Path");
-  else
-   dev_warn(&h->pdev->dev,
-    "%s: Error 0x%02x, retrying on standard path.\n",
-    "HP SSD Smart Path", c2->error_data.status);
-
   dev->offload_enabled = 0;
   h->drv_req_rescan = 1; /* schedule controller for a rescan */
   cmd->result = DID_SOFT_ERROR << 16;
-- 
1.7.1